File: objc/Cake ListTests/Cake_ListTests.m
//
//  Cake_ListTests.m
//  Cake ListTests
//
//  Created by Stewart Hart on 19/05/2015.
//  Copyright (c) 2015 Stewart Hart. All rights reserved.
//

#import "CakeData.h"

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>


@interface Cake_ListTests : XCTestCase

@end

@implementation Cake_ListTests

- (void)setUp {
    [super setUp];
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testExample {
    // This is an example of a functional test case.
    XCTAssert(YES, @"Pass");
}


- (void)testData {
    NSData *data = [CakeData getData];
    
    XCTAssert(data != nil, @"CakeData getData is nil");
    
    id response = [CakeData parseData:data];

    XCTAssert([response isKindOfClass:[NSArray class]], @"CakeData parseData returns error:%@",(NSError*)response);
}


- (void)testPerformanceExample {
    // This is an example of a performance test case.
    [self measureBlock:^{
        // Put the code you want to measure the time of here.
        [self testData];
    }];
}

@end

File: swift/CakeList/CakeListTests/CakeListTests.swift
//
//  CakeListTests.swift
//  CakeListTests
//
//  Created by Jamie Lemon on 19/04/2019.
//  Copyright © 2019 dijipiji. All rights reserved.
//

import XCTest
@testable import CakeList

class CakeListTests: XCTestCase {

    override func setUp() {
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
    }

    func testExample() {
        // This is an example of a functional test case.
        // Use XCTAssert and related functions to verify your tests produce the correct results.
    }
    
    func testData() {
        
        let cakeData:Data? = CakeData.getData()
        
        XCTAssert(cakeData != nil, "CakeData.getData() returns nil")
        
        let json:Any? = CakeData.parseData(cakeData!)
        
        XCTAssert(json != nil, "CakeData.parseData() returns nil")
        
        print("json=\(String(describing: json))")
        
    }

    func testPerformanceExample() {
        // This is an example of a performance test case.
        self.measure {
            // Put the code you want to measure the time of here.
        }
    }

}
